core/rest/src/main/java/org/phenotips/remote/rest/internal/DefaultRemotePatientMatchResource.java

```diff
@@ -0,0 +1,214 @@
+    {
+        // Get the request container.
+        final Request request = this.container.getRequest();
+        // The patient ID must not be blank.
+        if (StringUtils.isBlank(patientId)) {
+            this.logger.error("Patient ID is not specified.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        // The server must not be blank.
+        final String server = (String) request.getProperty(SERVER);
+        if (StringUtils.isBlank(server)) {
+            this.logger.error("Server is not specified.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        // Get the other parameters, if specified, or set the defaults.
+        final int offset =  NumberUtils.toInt((String) request.getProperty(OFFSET), 1);
+        if (offset < 1) {
+            this.logger.error("The requested offset is out of bounds: {}", offset);
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        final boolean newRequest = BooleanUtils.toBoolean((String) request.getProperty(SEND_NEW_REQUEST));
+        final int limit = NumberUtils.toInt((String) request.getProperty(LIMIT), -1);
+        final int reqNo = NumberUtils.toInt((String) request.getProperty(REQ_NO), 1);
+        // Build the response.
+        return buildResponse(patientId, server, offset, limit, newRequest, reqNo);
+    }
+
+    /**
+     * Builds a response containing the reference patient and matched patients data, or an error code if unsuccessful.
+     *
+     * @param patientId the local reference patient identifier
+     * @param server the remote server that will be queried for matches
+     * @param offset the offset for the returned matches
+     * @param limit the maximum number of matches to return after the offset
+     * @param newRequest true iff a new match request should be made to the remote server
+     * @param reqNo the current request number
+     * @return a {@link Response} containing the matched patients data, or an error code if unsuccessful
+     */
+    private Response buildResponse(
+        @Nonnull final String patientId,
+        @Nonnull final String server,
+        final int offset,
+        final int limit,
+        final boolean newRequest,
+        final int reqNo)
+    {
+        try {
+            // Checks if the current user has access to the requested patient.
+            final Patient patient = this.repository.get(patientId);
+            // If patient with requested ID is not found, this is an error.
+            if (patient == null) {
+                this.logger.error("Patient with ID: {} could not be found.", patientId);
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            // FIXME: Magic number taken from PhenomeCentral.RemoteResults
+            final OutgoingMatchRequest remoteResponse = newRequest
+                ? this.matchingService.sendRequest(patientId, server, 5)
```

Comment:
according to [those lines](https://github.com/phenotips/remote-matching/blame/RM-59/ui/src/main/resources/PhenomeCentral/RemoteResults.xml#L58-L59) we should use 0 instead of 5, which is equivalent to: `final OutgoingMatchRequest remoteResponse = this.matchingService.getLastRequestSent(patientId, server);`